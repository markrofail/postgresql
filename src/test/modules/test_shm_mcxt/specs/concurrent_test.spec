setup
{
    CREATE EXTENSION test_shm_mcxt;
}

teardown
{
    DROP EXTENSION test_shm_mcxt;
}

session "s1"
step "add_s1_1" {CALL set_shared_list(1);}
step "remove_s1_1" {CALL set_shared_list(-1);}
step "remove_s1_2" {CALL set_shared_list(-2);}
step "get_s1" {SELECT * from get_shared_list();}

session "s2"
step "add_s2_2" {CALL set_shared_list(2);}
step "remove_s2_1" {CALL set_shared_list(-1);}
step "get_s2" {SELECT * from get_shared_list();}

permutation "add_s1_1" "add_s2_2" "get_s1" "remove_s1_2" "remove_s2_1" "get_s1"
permutation "add_s1_1" "get_s2" "remove_s1_2" "get_s1"

"add_s1_1" "remove_s1_1" "remove_s2_1"
