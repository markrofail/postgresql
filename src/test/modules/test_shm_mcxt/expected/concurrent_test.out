Parsed test spec with 2 sessions

starting permutation: add_s1_1 add_s2_2 get_s1 remove_s1_2 remove_s2_1 get_s1
step add_s1_1: CALL set_shared_list(1);
step add_s2_2: CALL set_shared_list(2);
step get_s1: SELECT * from get_shared_list();
get_shared_list

1              
2              
step remove_s1_2: CALL set_shared_list(-2);
step remove_s2_1: CALL set_shared_list(-1);
step get_s1: SELECT * from get_shared_list();
get_shared_list


starting permutation: add_s1_1 get_s2 remove_s1_2 get_s1 add_s1_1 remove_s1_1 remove_s2_1
step add_s1_1: CALL set_shared_list(1);
step get_s2: SELECT * from get_shared_list();
get_shared_list

1              
step remove_s1_2: CALL set_shared_list(-2);
step get_s1: SELECT * from get_shared_list();
get_shared_list

1              
step add_s1_1: CALL set_shared_list(1);
step remove_s1_1: CALL set_shared_list(-1);
step remove_s2_1: CALL set_shared_list(-1);
